custom_components/tuya_local/devices/avidsen_soria_solarinverter.yaml

```diff
@@ -0,0 +1,183 @@
+name: Solar kit 400W
+products:
+  - id: 0kllybtbzftaee7y
+    name: Karst-400
+primary_entity:
+  entity: sensor
+  class: energy
+  dps:
+    - id: 2
+      type: integer
+      name: sensor
+      unit: kWh
+      class: total_increasing
+      mapping:
+        - scale: 100
+    - id: 15
+      name: inverter_type
+      type: string
+    - id: 16
+      name: inverter_id
+      type: string
+    - id: 17
+      name: imei_imsi
+      type: string
+secondary_entities:
+  - entity: switch
+    name: Inverter switch
+    category: config
+    dps:
+      - id: 101
+        type: boolean
+        name: switch
+        # force: true
+  - entity: number
+    name: Power Adjustement
+    category: config
+    class: power_factor
+    dps:
+      - id: 105
+        type: integer
+        name: value
+        range:
+          min: 0
+          max: 100
+        unit: "%"
+  - entity: button
+    name: Reset energy
+    category: config
+    class: restart
+    dps:
+      - id: 104
+        type: string
+        name: button
+        optional: true
+        mapping:
+          - dps_val: "1"
+            value: true
+  - entity: sensor
+    name: AC Power
+    class: power
+    dps:
+      - id: 103
+        type: integer
+        name: sensor
+        unit: W
+        force: true
```

Comment:
DP is not optional, does it really require force? Normally DP that require force will not show up by default, so need to also be optional.

Reply:
OK, I wasn't sure about that. I've changed it in my HA instance, but device has just gone offline because of night coming. I'll check tomorrow it is OK.

Reply:
Tested and OK, I also checked again device recognicion.